### DISCUSSION TITLE
Changing subscriptions results in flickering with experimental bag player

### DISCUSSION THREAD

**Discussion by jtbandes** [foxglove/community#534] [Q&A]
When using the experimental bag player, changing subscriptions (e.g. typing in the message path input in a Raw Message panel) causes existing panels to flicker as they re-backfill their data.

Video shows first the old bag player, and then the new experimental player:

https://user-images.githubusercontent.com/14237/160724810-29b78ba9-1d53-4027-8710-e3b5186ef544.mov

**Comment by defunctzombie** [comment]
I think its because of this line: https://github.com/foxglove/studio/blob/main/packages/studio-base/src/players/IterablePlayer/IterablePlayer.ts#L319
